operators/pkg/controller/elasticsearch/version/common.go

```diff
@@ -181,10 +182,14 @@ func podSpec(
 	// we do not override resource Requests here in order to end up in the qosClass of Guaranteed by default
 	// see https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/ for more details
 
-	// augment user-provided env vars with our own
-	// TODO: deal with conflicts here (eg. JVM_OPTIONS)
 	heapSize := MemoryLimitsToHeapSize(*containerSpec.Resources.Limits.Memory())
-	containerSpec.Env = append(containerSpec.Env, newEnvironmentVarsFn(p, heapSize, httpCertificatesVolume, reloadCredsSecret, secureSettingsVolume)...)
+	// inherit user-provided environment...
+	envBuilder := overrides.NewEnvBuilder(containerSpec.Env...)
```

Comment:
I completely agree with the naming. FWIW I think I would have chosen `env.NewBuilder`. We will see in the future if we add new 'overridable' things or rather new 'env related' things.

Reply:
Yeah good point.
What I had in mind was to have override helpers for:
* environment variables
* labels (turns out not required since it's already a map hence easy to use directly)
* annotations
* volumes
* volume mounts